src/components/Markets/index.tsx

```diff
@@ -123,52 +62,71 @@ export class Markets extends React.Component<MarketsProps, MarketsState> {
+    }, [renderChange]);
 
-    private filterType = (headerKey: string, searchKey: string) => (item: CellData[]) => {
-        const typeIndex = this.props.headers ? this.props.headers.indexOf(headerKey) : this.defaultHeaders.indexOf(headerKey);
+    const filterType = React.useCallback((headerKey: string, key: string) => (item: CellData[]) => {
+        const typeIndex = (props.headers || DEFAULT_MARKET_HEADERS).indexOf(headerKey);
 
-        return (item[typeIndex] as string).includes(searchKey);
-    };
+        return (item[typeIndex] as string).includes(key);
+    }, [props.headers]);
 
-    private get filters() {
-        const { data } = this.props;
+    const createUniqueCurrencies = React.useCallback((currencies: string[], market: string) => {
+        const marketCurrencies = market.split('/').map((c: string) => c.trim());
+        const uniqueCurrencies = marketCurrencies.filter(c => !hasDuplicates(currencies, c));
+
+        return currencies.concat(uniqueCurrencies);
+    }, []);
 
+    const transformCurrencyToFilter = React.useCallback((currency: string) => ({
+        name: currency,
+        filter: filterType('Pair', currency),
+    }), [filterType]);
+
+    const getFilters = React.useCallback(() => {
         const currencyFilters = data && data.length > 0
-            ? this.props.data
-                .map(this.getMarketFromDataRow)
-                .reduce(this.createUniqueCurrencies, [])
-                .map(this.transformCurrencyToFilter)
+            ? data
+                .map((market: React.ReactNode[]) => market[0] as string)
+                .reduce(createUniqueCurrencies, [])
+                .map(transformCurrencyToFilter)
             : [];
 
-        return [
+        return filters ? [
             {
                 name: 'All',
-                filter: this.filterType('Pair', ''),
+                filter: filterType('Pair', ''),
             },
             ...currencyFilters,
-        ];
-    }
-
-    private getMarketFromDataRow = (market: React.ReactNode[]) => market[0] as string;
-
-    private createUniqueCurrencies(currencies: string[], market: string) {
-        const isCurrencyUnique = (currency: string) => !currencies.includes(currency);
-
-        const marketCurrencies = market.split('/').map((c: string) => c.trim());
-        const uniqueCurrencies = marketCurrencies.filter(isCurrencyUnique);
-
-        return currencies.concat(uniqueCurrencies);
-    }
-
-    private transformCurrencyToFilter = (currency: string) => ({
-        name: currency,
-        filter: this.filterType('Pair', currency),
-    });
-}
+        ] : [];
+    }, [createUniqueCurrencies, filterType, transformCurrencyToFilter, data, filters]);
+
+    const getTableData = React.useCallback(() => {
+        const fd = data.filter(w => (w[0] as string).toLowerCase().includes(searchKey.toLowerCase()));
+
+        return fd.map(row => row.map(mapRows));
+    }, [data, mapRows, searchKey]);
```

Comment:
1. Wrong name of method. Looks like here rather filter but not ```getTableData```
2. This looks to complex. Can you simplify code?